### PR TITLE
system_firmware conversion complete

### PR DESCRIPTION
**Pull request by sbeaver-netgate**
Ready for review

### CODE REVIEW

File: usr/local/www/system_firmware.php [hunk 1]
```diff
@@ -202,116 +197,128 @@ function file_upload_error_message($error_code) {
...
-<?php if ($sig_warning && !$input_errors): ?>
-<?php
-	$sig_warning = "<strong>" . $sig_warning . "</strong><br />" . gettext("This means that the image you uploaded " .
+if ($input_errors)
+	print_input_errors($input_errors);
+
+if ($savemsg)
+	print_info_box($savemsg);
+
+if ($fwinfo <> "")
```
Comment (SjonHortensius) [inline usr/local/www/system_firmware.php]: Cleaner should have converted this to `!=`

File: usr/local/www/system_firmware.php [hunk 2]
```diff
@@ -102,14 +102,9 @@ function file_upload_error_message($error_code) {
...
 	include("head.inc");
-	echo "<body link=\"#0000CC\" vlink=\"#0000CC\" alink=\"#0000CC\">\n";
 	include("fbegin.inc");
-	echo "<div>\n";
-	print_info_box(gettext("An upgrade is currently in progress.<p>The firewall will reboot when the operation is complete.") . "</p><p><img src='/themes/{$g['theme']}/images/icons/icon_fw-update.gif' alt='update' /></p>");
-	echo "</div>\n";
-	include("fend.inc");
-	echo "</body>";
-	echo "</html>";
+	print_info_box(gettext("An upgrade is currently in progress. The firewall will reboot when the operation is complete.") . "<p><img src='/themes/{$g['theme']}/images/icons/icon_fw-update.gif' alt='update' /></p>");
```
Comment (SjonHortensius) [inline usr/local/www/system_firmware.php:106]: Is this image still present in the docroot? I'd guess it's no longer necessary

Comment (sbeaver-netgate) [inline usr/local/www/system_firmware.php:106]: Yes it is. It is a rather nice animated image so I left it in place. We can always remove it later

> On Apr 30, 2015, at 3:41 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/system_firmware.php https://github.com/SjonHortensius/pfsense/pull/157#discussion_r29462803:
> 
> > ```
> > include("fbegin.inc");
> > ```
> > -   echo "<div>\n";
> > -   print_info_box(gettext("An upgrade is currently in progress.<p>The firewall will reboot when the operation is complete.") . "</p><p>update</p>");
> > -   echo "</div>\n";
> > -   include("fend.inc");
> > -   echo "</body>";
> > -   echo "</html>";
> > -   print_info_box(gettext("An upgrade is currently in progress. The firewall will reboot when the operation is complete.") . "<p>update</p>");
> >   Is this image still present in the docroot? I'd guess it's no longer necessary
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/157/files#r29462803.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

File: usr/local/www/system_firmware.php [hunk 3]
```diff
@@ -202,116 +197,128 @@ function file_upload_error_message($error_code) {
...
-</html>
+
+} else {
+	// This is where the work gets done so Forms.classes will be used from this point
+	if (!is_subsystem_dirty('firmwarelock')) {
+		require('classes/Form.class.php');
+
+		if (!is_subsystem_dirty('rebootreq')) {
+			if (!is_subsystem_dirty('firmware')) {
+				$enablebtn = new Form_Button(Submit, 'Enable firmware upload', null);
```
Comment (SjonHortensius) [inline usr/local/www/system_firmware.php]: Please use proper newlines here. Also, _Submit_ should be quoted as its a string. And the 3rd parameter is null by default, so please omit it

**Pull request merged**